contracts/PodCastNFT.sol

```diff
@@ -225,9 +203,63 @@ contract PodCastNFT is ERC721URIStorage, Ownable {
         require(success, "Transaction Unsuccessful");
     }
 
-    function burn(
-        uint256 _tokenId
-    ) public onlyOwner {
+    function burn(uint256 _tokenId) public onlyOwner {
         _burn(_tokenId);
     }
+
+    function setKeyword(string memory _keyword, uint256 startedAt)
+        public
+        onlyOwner
+    {
+        weeklyKeyword = PodcastKeyword(
+            startedAt,
+            keccak256(abi.encodePacked(_keyword))
+        );
+    }
+
+    function getUserAttributes(address _of)
+        public
+        view
+        returns (Attributes memory)
+    {
+        return userAttribute[_of];
+    }
+
+    function checkAnswer(string memory _keyword)
+        public
+        onlyHolder(msg.sender)
+        returns (bool)
+    {
+        bool isCorrect = weeklyKeyword.keyword ==
+            keccak256(abi.encodePacked(_keyword));
+        if (!isCorrect) {
+            return false;
+        }
+        require(
+            userAttribute[msg.sender].answeredAt <= weeklyKeyword.startedAt,
+            "You cannot answer twice"
+        );
+        userAttribute[msg.sender].point += 100;
+        userAttribute[msg.sender].claimableToken += 100e18;
+        userAttribute[msg.sender].answeredAt = block.timestamp;
+        return true;
```

Comment:
There isn't really a point for non-view functions to return a value as you can't get the value until the transaction is confirmed. The usual way is to call separate view function or emit Event and get the info via subgraph

Reply:
probably you could call `answer` than `checkAnswer` to make it clear that it's not a view function